deployment/common/changeset/set_config_mcms.go

```diff
@@ -0,0 +1,207 @@
+}
+
+// setConfigOrTxData executes set config tx or gets the tx data for the MCMS proposal
+func setConfigOrTxData(chain deployment.Chain, cfg config.Config, contract *gethwrappers.ManyChainMultiSig, useMCMS bool) (*types.Transaction, error) {
+	groupQuorums, groupParents, signerAddresses, signerGroups := cfg.ExtractSetConfigInputs()
+	opts := deployment.SimTransactOpts()
+	if !useMCMS {
+		opts = chain.DeployerKey
+	}
+	tx, err := contract.SetConfig(opts, signerAddresses, signerGroups, groupQuorums, groupParents, false)
+	if err != nil {
+		return nil, err
+	}
+	if !useMCMS {
+		_, err = deployment.ConfirmIfNoError(chain, tx, err)
+		if err != nil {
+			return nil, err
+		}
+	}
+	return tx, nil
+}
+
+type setConfigTxs struct {
+	proposerTx  *types.Transaction
+	cancellerTx *types.Transaction
+	bypasserTx  *types.Transaction
+}
+
+// setConfigPerRole sets the configuration for each of the MCMS contract roles on the mcmsState.
+func setConfigPerRole(chain deployment.Chain, cfg ConfigPerRole, mcmsState *MCMSWithTimelockState, useMCMS bool) (setConfigTxs, error) {
+	// Proposer set config
+	proposerTx, err := setConfigOrTxData(chain, cfg.Proposer, mcmsState.ProposerMcm, useMCMS)
+	if err != nil {
+		return setConfigTxs{}, err
+	}
+	// Canceller set config
+	cancellerTx, err := setConfigOrTxData(chain, cfg.Canceller, mcmsState.CancellerMcm, useMCMS)
+	if err != nil {
+		return setConfigTxs{}, err
+	}
+	// Bypasser set config
+	bypasserTx, err := setConfigOrTxData(chain, cfg.Bypasser, mcmsState.BypasserMcm, useMCMS)
+	if err != nil {
+		return setConfigTxs{}, err
+	}
+
+	return setConfigTxs{
+		proposerTx:  proposerTx,
+		cancellerTx: cancellerTx,
+		bypasserTx:  bypasserTx,
+	}, nil
+}
+
+func addTxsToProposalBatch(setConfigTxsChain setConfigTxs, chainSelector uint64, state MCMSWithTimelockState) timelock.BatchChainOperation {
+	result := timelock.BatchChainOperation{
+		ChainIdentifier: mcms.ChainIdentifier(chainSelector),
+		Batch:           []mcms.Operation{},
+	}
+	result.Batch = append(result.Batch, mcms.Operation{
+		To:           state.ProposerMcm.Address(),
+		Data:         setConfigTxsChain.proposerTx.Data(),
+		Value:        big.NewInt(0),
+		ContractType: string(commontypes.ProposerManyChainMultisig),
+	})
+	result.Batch = append(result.Batch, mcms.Operation{
+		To:           state.CancellerMcm.Address(),
+		Data:         setConfigTxsChain.cancellerTx.Data(),
+		Value:        big.NewInt(0),
+		ContractType: string(commontypes.CancellerManyChainMultisig),
+	})
+	result.Batch = append(result.Batch, mcms.Operation{
+		To:           state.BypasserMcm.Address(),
+		Data:         setConfigTxsChain.bypasserTx.Data(),
+		Value:        big.NewInt(0),
+		ContractType: string(commontypes.BypasserManyChainMultisig),
+	})
+	return result
+}
+
+// SetConfigMCMS sets the configuration of the MCMS contract on the chain identified by the chainSelector.
+func SetConfigMCMS(e deployment.Environment, cfg SetConfigParams) (deployment.ChangesetOutput, error) {
```

Comment:
we should start making use of the context now that its available. ctx := e.GetContext() then you can pass it in to `opts.Context = ctx` when using the deployer key. Then if cancelled we'll respect it and gracefully stop

Reply:
added here, I'm setting it for both the deployerKeys and the SimTransactOps, I guess there should not be an issue setting it for both right? https://github.com/smartcontractkit/chainlink/pull/15684/commits/0f4cbf369a6f66423b89a2ffdd68add367691c1e